### PR TITLE
feat: es6 class should not be calling without 'new'

### PR DESCRIPTION
**Pull request by gxcsoccer**
add one restriction that es6 class should be calling directly without `new`.

### CODE REVIEW

**Comment by gxcsoccer** [comment]
change travis: only support node >= 2.0.0

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
+  - '4.2'
   - 'iojs-2'
```
Comment (fengmk2) [inline .travis.yml]:
```
- 4
- 3
- 2
- 1
- 0.12
```

File: package.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
     "loading"
   ],
   "engines": {
-    "node": ">= 0.10.0"
+    "node": ">= 2.0.0"
```
Comment (fengmk2) [inline package.json]: 呃。。。这个兼容一下还好吧？

**Comment by popomore** [comment]
It‘s a major change?

File: package.json (hunk 2 above)
Comment (gxcsoccer) [inline package.json]: 加了 isClass 0.10.0 下挂了，或者判断版本低就skip那个case?

Comment (fengmk2) [inline package.json]: 加个版本判断，对的。

**Comment by gxcsoccer** [comment]
> It‘s a major change?
 minor

**Comment by gxcsoccer** [comment]
rebase done

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
1.10.0

**Comment by gxcsoccer** [comment]
tnpm publish ?

**Comment by gxcsoccer** [comment]
btw, now i think it is a major change...

**Comment by fengmk2** [comment]
Why is major change?

**Comment by gxcsoccer** [comment]
> Why is major change?

it is not compatible with es6 class or babel class.  old version will throw error or call the constructor directly, but new version will not call it

**Pull request merged**